### DISCUSSION TITLE
What does no solid infill over perimeters do?

### DISCUSSION THREAD

**Discussion by blex-max** [supermerill/SuperSlicer#2050] [Q&A] [answered]
Hi All,

I'm finding the tooltip for no solid infill over X perimeters very confusing, and I've never actually been able to work out what it does - can someone enlighten me? If it would be helpful I'd be happy to assist with rewriting tooltips into more natural English as I'm sure that's a time drain for the project.

Thanks!

**Comment by supermerill** [comment, marked as answer]
see #1090

**Reply by blex-max** [reply]
This doesn't actually answer the question. And am I to take it that assistance with writing documentation is not wanted?

**Reply by supermerill** [reply]
I'm totally okay with other writing doc & helping me with names.

here is some other issues that talk about the problem this setting solve:
#253
#426

**Reply by blex-max** [reply]
Hmm ok, I understand the problem it solves, but I still can't work out how it does this. If it would be ok, could you perhaps explain it to me in French and I will consult with my Francophone colleagues?

As for helping with doc, is that something I should create a pull request for? I'll try to give some time to it.

**Reply by supermerill** [reply]
It checks how many perimeters there are under it (or over it, depends if it checks for bottom or top shell). If there is enough then it removes the area from the list of "solid fill" and put it back into the list of "sparse infill".

In this picture:
![image](https://user-images.githubusercontent.com/6536403/146278398-afdcb5fb-46dd-4445-aaae-f99613e20134.png)
if you set this setting to 2, then it can add solid infill in the "1" area (if there is at least 2 bottom layers) but won't in the "2" area.

Je peux l'expliquer en français mais je ne saurais pas trop quoi raconter d'autre.

**Reply by blex-max** [reply]
Je comprends maintenant, merci beaucoup. Je pense que je peux le préciser dans les doc.

Excuse my bad French, and thank you for taking the time to explain this.

**Reply by blex-max** [reply]
May I create a pull request for editing the doc?

**Reply by supermerill** [reply]
yes
